internal/storage/inmem/store.go

```diff
@@ -0,0 +1,328 @@
+package inmem
```

Comment:
nit:

Do you think it would be worthwhile to break apart this file into a few:

store_schema.go - for defining the memdb schema and constants (maybe this should be combined with query.go?)
store_resources.go - for methods pertaining to the core resource centric method
store_monotonic_index.go - for methods pertaining to maintaining the monotonically increasing index counter.